library/ecp.c

```diff
@@ -582,11 +582,14 @@ void mbedtls_ecp_group_free(mbedtls_ecp_group *grp)
     }
 
     if (grp->h != 1) {
-        mbedtls_mpi_free(&grp->P);
         mbedtls_mpi_free(&grp->A);
         mbedtls_mpi_free(&grp->B);
         mbedtls_ecp_point_free(&grp->G);
-        mbedtls_mpi_free(&grp->N);
+        if (grp->h == 0) {
```

Comment:
pkparse test suite [will allocate 8 bytes](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/suites/test_suite_pkparse.function#L127) which need to be freed, during the negative tests of `Key ASN1 (ECPrivateKey, empty parameters)` . 

`mbedtls_ecp_group_free()` will be invoked as part of [`mbedtls_pk_free()`](https://github.com/Mbed-TLS/mbedtls/blob/development/library/pk.c#L68), and it is expected to clean up the allocated memory appropriately.

Reply:
Also, we are changing the behaviour here and should update the documentation of `h`:
https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-3.3.0/include/mbedtls/ecp.h#L243

Reply:
`pk_group_id_from_group()` has been updated to free() and load() the temporary group as described in the above comment